src/cloud/components/Modal/contents/SmartView/SmartViewForm.tsx

```diff
@@ -37,19 +38,32 @@ interface SmmartViewFormProps {
 const SmartViewForm = ({
   action,
   defaultName = '',
-  defaultPrivate = true,
   defaultConditions,
   buttonsAreDisabled,
   showOnlyConditions,
+  defaultViewType,
   teamId,
   onCancel,
   onSubmit,
-}: SmmartViewFormProps) => {
+}: SmartViewFormProps) => {
   const [name, setName] = useState(defaultName)
-  const [makingPrivate, setMakingPrivate] = useState(defaultPrivate)
   const { translate } = useI18n()
 
   const [conditions, setConditions] = useState<EditableQuery>(defaultConditions)
+  const [viewType, setViewType] = useState<FormSelectOption | undefined>(
+    defaultViewType != null
+      ? getFormOptionFromViewType(defaultViewType)
+      : undefined
+  )
+
+  const viewTypeOptions: FormSelectOption[] = useMemo(() => {
```

Comment:
Considering have the value outside of the function component as long as it isn't relying on any props or state.